server/container_create.go

```diff
@@ -393,6 +393,11 @@ func (s *Server) getAppArmorProfileName(annotations map[string]string, ctrName s
 	}
 
 	if profile == apparmor.ProfileRuntimeDefault {
```

Comment:
No. `--apparmor-profile` (the configuration option) is setting the default profile name for all `cri-o` containers. This code is loading **the wrong profile** with **the wrong name** and the actual profile won't be used.